packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -162,10 +163,66 @@ function* pollForPurchaseConfirmation({
   })
 }
 
+/** Attempts to purchase the requested amount of USDC, will throw on cancellation or failure */
+function* purchaseUSDC({ amount }: { amount: BNUSDC }) {
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const isBuyUSDCViaSolEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.BUY_USDC_VIA_SOL
+  )
+  const roundedAmount = ceilingBNUSDCToNearestCent(amount)
+    .div(BN_USDC_CENT_WEI)
+    .toNumber()
+
+  let result: RaceStatusResult | null = null
+  if (isBuyUSDCViaSolEnabled) {
+    yield* put(
+      buyCryptoViaSol({
+        // expects "friendly" amount, so dollars
```

Comment:
Note for later: This threw me off as the Stripe saga expects the value in cents and the buyCryptoViaSol sagas want it in dollars. Need to be careful when updating this code to use FixedDecimal to ensure that we don't pass the wrong order of magnitude. Likely we want all of the sagas to accept a `USDC` and convert it internally.
cc @rickyrombo

Reply:
what do you think of `USDC(BigInt(1000000)).toNumber() // 1.0` as a proposal? Right now the way to do this would be `Number(USDC(BigInt(1000000)).toString())`